packages/edit-site/src/components/homepage-redirect/index.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useHistory } from '../routes';
+import { store as editSiteStore } from '../../store';
+
+export default function HomepageRedirect() {
+	const params = useSelect(
+		( select ) => select( editSiteStore ).getHomepageParams(),
```

Comment:
This is an awesome improvement!

I'm wondering though, if we could select it upfront before rendering the React app like you mentioned. We can use `resolveSelect` and await for the promise to resolve before rendering the app. As long as the request is properly preloaded, I think the promise should resolve in just a few ticks. And then we won't need the loading state either. We still need this selector in React though for client side navigations.

Reply:
Yeah, I figured this should be possible, but I needed a bit more time. I managed to get it working with the latest commit though.

I went with using `apiFetch` for most of the api calls, as a resolver/selector combo feels unnecessarily complicated.

What do you think? It could probably use a bit of refining and error handling at the moment, and the preloading is still not working for `wp/v2/settings` for some reason.

Reply:
LGTM 👍 ! This feels like the right approach, at least until #36873 😜 .